### ISSUE TITLE
Could not connect to "localhost:9009"

### ISSUE THREAD

**Issue by Sharaddition** [questdb/py-questdb-client#21]
After writing approx 16,000 rows in multiple tables I can no longer write data.
Error -

> Could not connect to "localhost:9009": Only one usage of each socket address (protocol/network address/port) is normally permitted. (os error 10048)

A few days back I didn't have any such issue, I wrote millions of rows in a similar fashion. My program is writing data in QuestDB from 2 WebSockets running in parallel processes.

**Comment by amunra** [comment]
Hi @Sharaddition, would you be able to give me more insight into your setup?
The client does not use websockets, so I would need some more clarity here.
Do you have reconnecting logic?

A few things to take notice:
* The server will disconnect once there is a data error. See: https://py-questdb-client.readthedocs.io/en/latest/troubleshooting.html#errors-during-flushing
* The server will no longer accept new concurrent TCP connections beyond a certain active connection count. It will stop the listening socket until the the active connection count goes down to the amount configurable by `line.tcp.net.connection.limit`. See: https://questdb.io/docs/reference/configuration/.

Get back to me (also on Slack) if you require further assistance.

**Comment by amunra** [comment]
@Sharaddition It could also be that you've exhausted the total usable set of outbound ports on the clients.
For that I'd need to set `SO_REUSEADDR` on the outbound socket (i.e. https://github.com/questdb/c-questdb-client/blob/main/questdb-rs/src/ingress/mod.rs#L1596).
It's pretty unusual to use up all outbound ports though, and it might be hiding another issue (see links above).

**Comment by amunra** [comment]
@Sharaddition 
Closing the issue due to inactivity. Feel free to re-open if you're still having issues or contact us on slack.
